Mathlib/Algebra/CharP/ExpChar.lean

```diff
@@ -111,3 +112,79 @@ end NoZeroDivisors
 end Nontrivial
 
 end Semiring
+
+theorem ExpChar.exists [Ring R] [NoZeroDivisors R] [Nontrivial R] :
+    ∃ q, ExpChar R q := by
+  obtain ⟨p, h⟩ := CharP.exists R
+  by_cases hp : p = 0
+  · exact ⟨1, by rw [hp] at h; haveI := CharP.charP_to_charZero R; exact .zero⟩
+  exact ⟨p, haveI := NeZero.mk hp; .prime (CharP.char_is_prime_of_pos R p).out⟩
+
+/-- If the algebra map `R →+* A` is injective then `A` has the same exponential characteristic
+as `R`. -/
+theorem expChar_of_injective_algebraMap {R A : Type*}
+    [CommSemiring R] [Semiring A] [Algebra R A] (h : Function.Injective (algebraMap R A))
+    (q : ℕ) [hR : ExpChar R q] : ExpChar A q := by
```

Comment:
Maybe, but we don't have `RingHom` version for `CharP` and `CharZero` either, so we should add all of them.

Reply:
I add `{charP|charZero|expChar}_of_injective_ringHom`. ~~Seems that the original `algebraMap` version becomes useless, since both versions accept the same input argument `h : Function.Injective ...`, so `..._of_injective_algebraMap h` can be changed directly to `..._of_injective_ringHom h`.~~ No, the original version is still useful for `..._of_injective_algebraMap (by ...)` which allows type guessing.

Reply:
Thanks!